src/value.rs

```diff
@@ -215,6 +217,40 @@ impl Uninitialized for Value {
     }
 }
 
+impl FromGlibPtrNone<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut gobject_ffi::GValue) -> Self {
+        assert!(!ptr.is_null());
+        Value(ptr::read(ptr))
+    }
+}
+
+impl FromGlibPtrFull<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut gobject_ffi::GValue) -> Self {
+        let value = Value(ptr::read(ptr));
+        gobject_ffi::g_value_unset(ptr);
+        value
+    }
+}
+
+impl FromGlibPtrNone<*const gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *const gobject_ffi::GValue) -> Self {
+        assert!(!ptr.is_null());
+        Value(ptr::read(ptr))
+    }
+}
+
+impl FromGlibPtrFull<*const gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *const gobject_ffi::GValue) -> Self {
```

Comment:
A `*const GValue` seems wrong here. C code is never going to mark a pointer a `const` if you're going to own it, as otherwise all mutating functions (and free(), g_value_unset(), etc) would cause a compiler warning (you can't pass const pointers to them).